### PR TITLE
K8SPXC-758: Allow skipping TLS verification for backup storage

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPXC-758](https://jira.percona.com/browse/K8SPXC-758) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPXC-758

Add `verifyTLS` variable to `backup.storages`

### CODE REVIEW

File: pkg/pxc/backup/job.go [hunk 1]
```diff
@@ -46,6 +46,11 @@ func (bcp *Backup) JobSpec(spec api.PXCBackupSpec, cluster api.PerconaXtraDBClus
 
 	manualSelector := true
 	backbackoffLimit := int32(10)
+	verifyTLS := "1"
+	if cluster.Backup.Storages[spec.StorageName].VerifyTLS != nil &&
+		!*cluster.Backup.Storages[spec.StorageName].VerifyTLS {
+		verifyTLS = "0"
+	}
```
Comment (delgod) [inline pkg/pxc/backup/job.go:53]: can we keep it simple?
```suggestion
	verifyTLS := true
	if cluster.Backup.Storages[spec.StorageName].VerifyTLS != nil {
		verifyTLS = *cluster.Backup.Storages[spec.StorageName].VerifyTLS
	}
```

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by hors — CHANGES_REQUESTED** [review]
@pooknull please add new option into` deploy/cr.yaml `

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| recreate | passed |
| big-data | passed |
| scaling | passed |
| upgrade-haproxy | passed |
| storage | passed |
| scheduled-backup | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| self-healing-chaos | failed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| security-context | failed |
| upgrade-proxysql | passed |
| limits | passed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| pitr | passed |
| auto-tuning | passed |
| upgrade-consistency | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | failed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1077/commits/28f17bdb4cbfe3c380c6868f01be0e3ca1cad4db
image: `perconalab/percona-xtradb-cluster-operator:PR-1077-28f17bdb`

**Comment by pooknull** [comment]
@hors fixed in 5792a2c8bfa6e2b95d0612dcc8c7a86e13c2044d

File: pkg/pxc/backup/job.go (hunk 1 above)
Comment (pooknull) [reply pkg/pxc/backup/job.go]: Resolved in 9fbab09b01d7c2b1b6b840fe02ce350b06633618

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| upgrade-haproxy | passed |
| haproxy | passed |
| scaling | passed |
| storage | passed |
| recreate | passed |
| scheduled-backup | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| self-healing-chaos | failed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| security-context | failed |
| upgrade-proxysql | passed |
| limits | passed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| pitr | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| users | failed |
| demand-backup-encrypted-with-tls | failed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1077/commits/9fbab09b01d7c2b1b6b840fe02ce350b06633618
image: `perconalab/percona-xtradb-cluster-operator:PR-1077-9fbab09b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| storage | passed |
| upgrade-haproxy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| self-healing-chaos | failed |
| security-context | failed |
| scheduled-backup | passed |
| haproxy | passed |
| recreate | passed |
| big-data | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| pitr | passed |
| affinity | passed |
| demand-backup-encrypted-with-tls | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1077/commits/9fbab09b01d7c2b1b6b840fe02ce350b06633618
image: `perconalab/percona-xtradb-cluster-operator:PR-1077-9fbab09b`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| haproxy | passed |
| scaling | passed |
| storage | passed |
| scheduled-backup | passed |
| big-data | passed |
| recreate | passed |
| self-healing | passed |
| self-healing-chaos | failed |
| scaling-proxysql | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| security-context | failed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-proxysql | passed |
| users | passed |
| smart-update | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | failed |
| upgrade-consistency | failed |
| operator-self-healing | passed |
| operator-self-healing-chaos | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1077/commits/5792a2c8bfa6e2b95d0612dcc8c7a86e13c2044d
image: `perconalab/percona-xtradb-cluster-operator:PR-1077-5792a2c8`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| storage | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| recreate | passed |
| haproxy | passed |
| scheduled-backup | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| init-deploy | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| limits | passed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| affinity | passed |
| one-pod | passed |
| pitr | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1077/commits/aa096886ee5b8b21b7f42aad02cf269aa0ff00c1
image: `perconalab/percona-xtradb-cluster-operator:PR-1077-aa096886`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| operator-self-healing | passed |
| upgrade-haproxy | passed |
| operator-self-healing-chaos | passed |
| haproxy | passed |
| scheduled-backup | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| big-data | passed |
| affinity | passed |
| recreate | passed |
| scaling-proxysql | passed |
| one-pod | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| auto-tuning | passed |
| demand-backup | passed |
| proxysql-sidecar-res-limits | passed |
| pitr | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1077/commits/aa096886ee5b8b21b7f42aad02cf269aa0ff00c1
image: `perconalab/percona-xtradb-cluster-operator:PR-1077-aa096886`

**Review by delgod — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Pull request merged**